tools/2024-06-gitops-feature-spec.md

```diff
@@ -0,0 +1,374 @@
+
+<!-- What is the positive outcome for the customer if we deliver these features, i.e. what is the value proposition for the customer? Remember, this is customer-centric. -->
+
+## Key scenarios
+<!-- List ~3-7 high level scenarios to clarify the value of the Epic and point to how we will decompose this big area into component capabilities. We may ultimately have more than one level of scenario. -->
+
+### Scenario 1: As an operator I can install and configure Radius with my GitOps tool of choice
+<!-- One or two sentence summary -->
+My team uses GitOps to manage Kubernetes clusters and applications that run in those clusters.  All relevant configuration information is stored in git repositories that serve as the sources of truth for both the application code and infrastructure configuration.  To use Radius, I need to include Radius as a dependency for my Kubernetes cluster, just as I would any other dependency, via my GitOps tools.
+
+### Scenario 2: As an operator or developer, I can deploy and manage Radius Environments, Recipes, applications, resources, credentials, etc. via my GitOps toolset
+<!-- One or two sentence summary -->
+With Radius installed, as an Operator, I now define, deploy and manage a Radius environment and associate relevant Recipes, Credentials, and other Radius objects to that environment. To accomplish this, I just need to push my Radius Environment, Recipe, and other files into my git repository. Flux or ArgoCD will detect the changes and trigger Radius to deploy the Environment as specified.
+
+As a Developer, my Operations counterpart has provided a Radius Environment plus Recipes to enable my Radius application development. After I have familiarized myself with Radius, I push my application definition files into my git repository, where Flux or ArgoCD will detect the changes and trigger Radius to deploy the application to this environment using the Recipes. After my app is deployed, I can view its graph in the Radius dashboard. Everything is working well for me so far and I really appreciate these cool Radius features!
+
+### Scenario 3: As a Site Reliability Engineer (SRE), I can edit and customize the configurations in the Kubernetes cluster through my GitOps toolset, even after the application has been previously deployed.
+<!-- One or two sentence summary -->
+As an SRE, my Operations counterpart has provided a Kubernetes cluster, a Radius Environment running on that cluster and the infrastructure resources through Radius Recipes required for the Radius Applications that will run in that Radius Environment. I am expected to make adjustments to the configuration of the Kubernetes cluster and the Radius Environment as needed. I can do this through my GitOps toolset by updating the Bicep parameter files and pushing them into the git repository. Flux or ArgoCD will detect the changes and trigger Radius to redeploy the resources with the new configurations.
```

Comment:
How would the changes to the recipes be picked up by GitOps? Do we need to create a newer version of the recipe and update the environment so that it can point to the new recipe? How will we tell GitOps to do that?

Reply:
We should not handle the publishing step of recipes, i.e. `rad bicep publish`. Changes to the environment (e.g. recipe image tag) would be built into the definition of `env.bicep` and thus applied by GitOps as usual.

Reply:
infra operators (and/or automation that Radius users build) will be responsible for this step instead.

Reply:
added the clarification as a note